reactor3/src/test/java/com/linecorp/armeria/common/reactor3/RequestContextPropagationFluxTest.java

```diff
@@ -0,0 +1,473 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.reactor3;
+
+import static com.linecorp.armeria.common.reactor3.ContextAwareMonoTest.ctxExists;
+import static com.linecorp.armeria.common.reactor3.ContextAwareMonoTest.newContext;
+import static com.linecorp.armeria.common.reactor3.ContextAwareMonoTest.noopSubscription;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.time.Duration;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.reactivestreams.Publisher;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.common.RequestContextAccessor;
+import com.linecorp.armeria.internal.testing.AnticipatedException;
+import com.linecorp.armeria.internal.testing.GenerateNativeImageTrace;
+
+import reactor.core.Disposable;
+import reactor.core.publisher.ConnectableFlux;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+import reactor.test.StepVerifier;
+import reactor.test.StepVerifierOptions;
+import reactor.util.context.Context;
+
+@GenerateNativeImageTrace
+class RequestContextPropagationFluxTest {
+
+    @BeforeAll
+    static void setUp() {
+        RequestContextPropagationHook.enable();
+    }
+
+    @AfterAll
+    static void tearDown() {
+        RequestContextPropagationHook.disable();
+    }
+
+    @Test
+    void fluxError() {
+        final ClientRequestContext ctx = newContext();
+        final Flux<Object> flux;
+
+        flux = addCallbacks(Flux.error(() -> {
+            // This is called twice. after publishOn() and verifyErrorMatches()
+            // After publishOn(), ctxExists(ctx) should be false.
+            // On the other hand, it should be True due to ContextPropagation.
+            return new AnticipatedException();
+        }).publishOn(Schedulers.single()), ctx);
+
+        StepVerifier.create(flux, initialReactorContext(ctx))
+                    .expectSubscriptionMatches(s -> ctxExists(ctx))
+                    .verifyErrorMatches(t -> ctxExists(ctx) && t instanceof AnticipatedException);
+    }
+
+    @Test
+    void fluxFromPublisher() {
+        final ClientRequestContext ctx = newContext();
+        final Flux<Object> flux;
+
+        flux = addCallbacks(Flux.from(s -> {
+            assertThat(ctxExists(ctx)).isTrue();
+            s.onSubscribe(noopSubscription());
+            s.onNext("foo");
+            s.onComplete();
+        }).publishOn(Schedulers.single()), ctx);
+
+        StepVerifier.create(flux, initialReactorContext(ctx))
```

Comment:
We don't usually subscribe with the Reactor context.
We are just doing `flux.subscribe(...)`,`flux.collect(...)`, etc.
Could you remove `initialReactorContext(ctx)` from the test cases?
We can also change not to check the ctx in `expectSubscriptionMatches` and `expectNextMatches`.

Reply:
Thanks for your explanation 🙇‍♂️ 
I removed `initialReactorContext(ctx)`. 
Also, i removed all codes fragment of `ctxExists(ctx)` in Stepverifier. (Not in `Flux`, `Mono`)!